pymodbus/logging.py

```diff
@@ -32,12 +32,15 @@ class Log:
     _logger = logging.getLogger(__name__)
 
     @classmethod
-    def apply_logging_config(cls, level=logging.WARNING):
-        """Apply basic logging configuration"""
-        logging.basicConfig(
-            format="%(asctime)s %(levelname)-5s %(module)s:%(lineno)s %(message)s",
-            datefmt="%H:%M:%S",
-        )
+    def apply_logging_config(
+        cls, level: int = logging.WARNING, **config_overrides: dict
+    ) -> None:
+        """Apply basic logging configuration."""
+        default_config: dict = {
+            "format": "%(asctime)s %(levelname)-5s %(module)s:%(lineno)s %(message)s",
+            "datefmt": "%H:%M:%S",
+        }
+        logging.basicConfig(**{**default_config, **config_overrides})
```

Comment:
This is a highly complex statement and not needed, resolve config_overrides by adding it to default_config first.

Reply:
Imo a dict merge and then a dict unpack is not complex.  What do you think of this then?

```python
config_kwargs = {**default_config, **config_overrides}
logging.basicConfig(**config_kwargs)
```

Reply:
Ugly but I can accept that.

Reply:
Alright, it's implemented!